### DISCUSSION TITLE
Can we benefit from switching libcudnn.so to cuDNN's version?

### DISCUSSION THREAD

**Discussion by DevilaN** [AbdBarho/stable-diffusion-webui-docker#300] [Ideas]
Reddit user reported that there is speed improvement when switching from python's version of libcudnn.so to actual file from CUDNN package for linux: https://www.reddit.com/r/StableDiffusion/comments/10fw843/397_its_with_a_4090_on_linux/ )
Seems like it is worth testing. Unfortunately I do not have knowledge of Python / CUDA things and maybe there is someone who will manage to test it quickly? If nobody check this out I'll try next week and I'll report here about what I've found.

**Comment by DevilaN** [comment]
Changed this library and I do not see any improvement. Seems like a dead end right now :(

**Comment by kelvie** [comment]
https://developer.download.nvidia.com/compute/redist/cudnn/v8.8.0/local_installers/11.8/cudnn-local-repo-ubuntu2204-8.8.0.121_1.0-1_amd64.deb

I extracted the .so files from this deb to `/data/cudnn`, and added this to the entrypoint:

```sh
  PYTORCH_PATH=$(python -c "import torch; print(torch.__path__[0])")
  cp /data/cudnn/lib* "$PYTORCH_PATH"/lib/
```

to override the cuda files, and on an RTX 4090, this went from 5 it/s to 30+ .

I think this is worth doing

**Reply by AbdBarho** [reply]
Perhaps worth noting: this will only make a difference if you have a card with a high compute capability (e.g 8.6+), it won't make any difference on older architectures.

**Comment by digger73** [comment]
I have just replaced the cudnn files as well, using a 4090 doubled the speed from 10it/s to 20it/s. I'm using an old CPU intel 6800k, so didn't expect the 30+ but still happy about the gain. 
So there are gains regarding this. There is also a post here: https://www.reddit.com/r/StableDiffusion/comments/y71q5k/4090_cudnn_performancespeed_fix_automatic1111/ describing it as well.
